handlers/namespaces.go

```
@@ -126,13 +126,11 @@ func ConfigValidationSummary(w http.ResponseWriter, r *http.Request) {
 
 	validationSummaries := []models.IstioValidationSummary{}
 	for _, ns := range nss {
+		// for multicluster validation summaries, all selected namespaces are sent in a request per all clusters, so ignore not found namespaces
 		istioConfigValidationResults, errValidations := business.Validations.GetValidationsForNamespace(r.Context(), cluster, ns)
-		if errValidations != nil {
-			log.Error(errValidations)
-			RespondWithError(w, http.StatusInternalServerError, errValidations.Error())
-			return
+		if errValidations == nil {
+			validationSummaries = append(validationSummaries, *istioConfigValidationResults.SummarizeValidation(ns, cluster))
```

I don't think we should ignore _all_ errors here. Doesn't the problem stem from the frontend sending a bad request to the backend? The frontend is asking for validations for namespaces it shouldn't. Is it possible to fix that on the frontend? Namespaces are cluster specific and we should know which namespaces exist on which clusters on the frontend and only send those namespaces in with the query param.

---

Then we should rework the frontend, the namespaces selection box, it is not filtering namespaces per cluster and showing all merged, and while selecting, it makes a list of namespaces strings.

---

Which is a scope of another task https://github.com/kiali/kiali/issues/5932

---

> In backend, anyway exiting the loop in case of error and returning that error was wrong.

Why was that wrong? If there's an issue getting validations shouldn't the backend return an error instead of silently fail? In this case, the frontend is giving the backend an invalid list of namespaces. Maybe we should treat authorization errors, where the user doesn't have access to a namespace that exists, as 404 so that we don't leak namespace names.

The backend can do an access check for the namespaces requested and filter out the ones that the user doesn't have access to. If the user provides namespaces that don't exist in the query param post filtering, the backend can either return a 400 or silently return results without those namespaces.

You're right, we should address the namespace selector separately in #5932 but I think even without the the frontend should have both 1. the list of selected namespaces 2. the list of namespaces per cluster. Before you send a request to the backend the frontend would need to filter out 2. with 1. instead of just using the list of selected namespaces and sending a request per cluster.

---

Looking at the requests in Josune's screenshot, you can see that the client is asking for all namespaces for each cluster. I agree with Nick, that's wrong and should cause an authorization error in the backend.  The client should have full AccessibleNamespace info for the user, and should be able to narrow the namespaces as needed. There are lots of places in the UI code, like in the graph appenders, that need to protect against this sort of back-end violation. Looking at this, I think it's a client-side fix.

---

Right, thanks for the comments, have fixed the frontend to filter selected namespaces per cluster before sending requests.